File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int k = numbers.length-1;
        while(i<k){
            int sum = numbers[i]+numbers[k];
            if(sum>target){
                k--;
                continue;
            }
            if(sum<target){
                i++;
                continue;
            }
            if(sum==target){
                int[] arr = new int[2];
                arr[0]=i+1;
                arr[1]=k+1;
                return arr;
            }

        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
from typing import List
from collections import defaultdict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        values = defaultdict(list)
        for j, i in enumerate(nums):
            if len(values[target-i])!=0:
                return [min(j, values[target-i][0]), max(j, values[target-i][0])]
            values[i].append(j)
        return []